### PR TITLE
driver: remove LXC and ECS driver documentation.

### PR DESCRIPTION
**Pull request by jrasell**
The website documentation has been removed as a result of previous deprecation.

The experimental remote task driver documentation has also been removed and the driver capability marked as deprecated.

Do not backport, destined for 1.9 only.
Jira: https://hashicorp.atlassian.net/jira/software/c/projects/NET/issues/NET-11167

### CODE REVIEW

File: website/data/docs-nav-data.json [hunk 1]
```diff
@@ -2004,10 +2004,6 @@
             "title": "Lightrun",
             "href": "/plugins/drivers/community/lightrun"
           },
-          {
-            "title": "LXC",
-            "href": "/plugins/drivers/community/lxc"
-          },
```
Comment (tgross) [inline website/data/docs-nav-data.json:2010]: Should we remove the `rkt` link while we're here? That's are dead and just redirects back to the plugins / community page.

File: website/content/docs/drivers/index.mdx [hunk 2]
```diff
@@ -15,7 +15,7 @@ own drivers without having to recompile Nomad. You can view the [plugin block][p
 documentation for examples on how to use the `plugin` block in Nomad's client configuration.
 See [docker][docker_plugin] for a more detailed and specific example.
 Keep in mind that even though all built-in drivers are plugins, Nomad
-remains a single binary and maintains backwards compatibility except with the `lxc` driver.
+remains a single binary and maintains backwards compatibility.
```
[the remark this answers is not in the document]
Comment (aimeeu) [reply website/content/docs/drivers/index.mdx:18]: "Keep in mind that even though all built-in drivers are plugins, Nomad remains a single binary and maintains backwards compatibility."

I agree with Tim's suggestion to clarify that the Nomad binary includes the default drivers.

---
Task drivers are pluggable. This gives you the flexibility to introduce your own drivers without having to recompile Nomad. You can view the [plugin block](https://developer.hashicorp.com/nomad/docs/configuration/plugin) documentation for examples on how to use the plugin block in Nomad's client configuration. See the Docker driver's [Client Requirements](https://developer.hashicorp.com/nomad/docs/drivers/docker#client-requirements) for a more detailed and specific example. 

The Nomad binary includes the default task drivers, so you do not have to download them separately.

**Review by aimeeu — APPROVED** [review]
Replied to Tim's comment

File: website/data/docs-nav-data.json (hunk 1 above)
Comment (jrasell) [reply website/data/docs-nav-data.json:2010]: Removed this and other rkt mentions within https://github.com/hashicorp/nomad/pull/24107/commits/fa5ab74bdaacf5a68bd4537866afd151651f66e2

File: website/content/docs/drivers/index.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrasell) [reply website/content/docs/drivers/index.mdx]: Updated the two offending paragraphs within https://github.com/hashicorp/nomad/pull/24107/commits/a0180465399d699bb0fd06500d4e86736383f646

**Comment by aimeeu** [comment]
@jrasell just occurred to me that we really should put in a redirect from the deleted content URLs

/nomad/plugins/drivers/remote/ section  -> /nomad/plugins/drivers/index.mdx
/nomad/plugins/drivers/community/lxc.mdx -> /nomad/plugins/drivers/community/index.mdx

I'm not sure how to do redirects in this framework and am looking into it.

Also a PR in hashicorp/integrations repo (/nomad/hashicorp dir) to remove lxc.hcl

**Pull request merged**